executor/show.go

```diff
@@ -996,6 +1006,58 @@ func (e *ShowExec) fetchShowWarnings(errOnly bool) error {
 	return nil
 }
 
+// fetchShowPumpStatus gets status of all pumps and fill them into e.rows.
+func (e *ShowExec) fetchShowPumpStatus() error {
+	registry, err := createRegistry(config.GetGlobalConfig().Path)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	nodes, _, err := registry.Nodes(context.Background(), node.NodePrefix[node.PumpNode])
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	for _, n := range nodes {
+		e.appendRow([]interface{}{n.NodeID, n.Addr, n.State, n.MaxCommitTS, utils.TSOToRoughTime(n.UpdateTS)})
+	}
+
+	return nil
+}
+
+// fetchShowDrainerStatus gets status of all drainers and fill them into e.rows.
+func (e *ShowExec) fetchShowDrainerStatus() error {
+	registry, err := createRegistry(config.GetGlobalConfig().Path)
```

Comment:
need close the registry https://github.com/pingcap/tidb-tools/blob/c71b1effdf1b05b0db12b9d8b544d035b910b74d/tidb-binlog/node/registry.go#L31

Reply:
Done. PTAL